include/exadg/poisson/overset_grids/driver.h

```diff
@@ -0,0 +1,70 @@
+/*  ______________________________________________________________________
+ *
+ *  ExaDG - High-Order Discontinuous Galerkin for the Exa-Scale
+ *
+ *  Copyright (C) 2021 by the ExaDG authors
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+#ifndef INCLUDE_EXADG_POISSON_OVERSET_GRIDS_DRIVER_H_
+#define INCLUDE_EXADG_POISSON_OVERSET_GRIDS_DRIVER_H_
+
+// ExaDG
+#include <exadg/functions_and_boundary_conditions/interface_coupling.h>
+#include <exadg/poisson/solver_poisson.h>
+
+namespace ExaDG
+{
+namespace Poisson
+{
+template<int dim, int n_components, typename Number>
+class DriverOversetGrids
+{
+public:
+  DriverOversetGrids(
+    MPI_Comm const &                                                        mpi_comm,
+    std::shared_ptr<ApplicationOversetGridsBase<dim, n_components, Number>> application);
+
+  void
+  setup();
+
+  void
+  solve();
+
+private:
+  static unsigned int const rank =
+    (n_components == 1) ? 0 : ((n_components == dim) ? 1 : dealii::numbers::invalid_unsigned_int);
```

Comment:
I don't think it is a good idea to introduce `rank` here. I would rather use the traits from deal.II.

Reply:
we can talk about this again once https://github.com/dealii/dealii/issues/13517 is resolved.